src/libs/antares/study/area/list.cpp

```diff
@@ -1212,6 +1212,42 @@ static bool AreaListLoadFromFolderSingleArea(Study& study,
     return ret;
 }
 
+void AreaList::ensureDataIsInitialized(Parameters& params, bool loadOnlyNeeded)
+{
+    AreaListEnsureDataLoadTimeSeries(this);
+    AreaListEnsureDataSolarTimeSeries(this);
+    AreaListEnsureDataWindTimeSeries(this);
+    AreaListEnsureDataHydroTimeSeries(this);
+    AreaListEnsureDataThermalTimeSeries(this);
+    AreaListEnsureDataRenewableTimeSeries(this);
```

Comment:
Make all these function private methods of class `AreaList`
- `AreaListEnsureDataLoadTimeSeries`
- ...
- `AreaListEnsureDataRenewableTimeSeries`
- `AreaListEnsureDataLoadPrepro`
- ...
- `AreaListEnsureDataThermalPrepro`

Reply:
You could also create a specific class for Prepro, and another one for TimeSeries